tests/test_modules/test_pmac/test_pmacchildpart.py

```diff
@@ -96,81 +93,150 @@ def test_validate(self):
         expected = 0.010166
         assert ret.value.duration == expected
 
-    def do_check_output(self, user_programs=None, turnaround=200000):
+    def do_check_output_quantized(self):
+        assert self.child.handled_requests.mock_calls[:4] == [
+            call.post('writeProfile', csPort='CS1', timeArray=[0.002],
+                      userPrograms=[8]), call.post('executeProfile'),
+            call.post('moveCS1', a=-0.1374875093687539, b=0.0,
+                      moveTime=1.0374875094),
+            call.post('writeProfile',
+                      a=pytest.approx(
+                          [-0.125, 0., 0.125, 0.25, 0.375, 0.5, 0.625,
+                           0.63749375, 0.63749375, 0.63749375, 0.63749375,
+                           0.625, 0.5, 0.375, 0.25, 0.125, 0., -0.125,
+                           -0.13748751]),
+                      b=pytest.approx(
+                          [0., 0., 0., 0., 0., 0., 0., 0.01231466, 0.04975248,
+                           0.05024752, 0.08768534, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1,
+                           0.1, 0.1]),
+                      csPort='CS1',
+                      timeArray=pytest.approx(
+                          [99950, 500250, 500250, 500250, 500250, 500250,
+                           500250, 100000, 101000, 1000, 101000, 100000, 500250,
+                           500250, 500250, 500250, 500250, 500250, 99950]),
```

Comment:
It's a shame we need this 1ms point in the middle of the turnaround, but I guess it's a by-product of the way we calculate the velocity

Reply:
Its a product of the way we extend the times in order to quantize. If there was no middle section then the 0 length section gets extended to 1 ms.

I think we could use a different quantize approach for 'pointy hats'. Initially, I did have this but it broke because if there was a mix of pointy and flat hats then the times got extended to different values. Having re-thought the time extension to preserve symmetry, I now believe I could adjust it to work for pointy hats too.

Reply:
Modified the quantize to preserve pointy hats. See the comment in VelocityProfile.quantize()